packages/blocks/src/api/serializer.js

```diff
@@ -354,7 +362,17 @@ export function serializeBlock( block, { isInnerBlocks = false } = {} ) {
 	}
 
 	const blockType = getBlockType( blockName );
-	const saveAttributes = getCommentAttributes( blockType, block.attributes );
+	let saveAttributes = getCommentAttributes( blockType, block.attributes );
+
+	if ( __experimentalExcludeAttributes ) {
+		saveAttributes = omit(
```

Comment:
Here there is a problem when attributes produce different html like css classes. With my tests in Heading and `textAlign`, a css class is added and produces:
```
<!-- wp:heading {"textAlign":"center"} -->
<h2 class="has-text-align-center">hello</h2>
<!-- /wp:heading -->
```
So we can't use the initial `saveContent` produced by the previous attributes, because it will result in skipping the unwanted attributes but will keep the html. By doing that when we copy we end up with:
```
<!-- wp:heading {"className":"has-text-align-center"} -->
<h2 class="has-text-align-center">hello</h2>
<!-- /wp:heading -->
```

So what we need to do here if we have to omit attributes is:
1. get the remaining attributes by using `block.attributes`
2. get `saveContent` again with the above result
3. recalculate `saveAttributes`

The above flow is for `copy` actions

Reply:
the serializer is probably not the right place to do this. it's going to cause a lot of unintended consequences elsewhere because of the mashing of concerns. we should be able to exclude the attributes when cloning and then we won't have to touch `serializeBlock` and we won't have any issues with the contradiction between the serialized markup and the block attributes.

Reply:
do we need this change at all given that it's added already in `cloneBlock()`?

Reply:
Thank you, this is a very good catch 😬 

This was done to support the `copy` actions, so that's:
* using the `Copy` item in the block [toolbar](https://github.com/WordPress/gutenberg/pull/38643/files#diff-9e90ff9d1914908a6c9c3b91a99ad894998ed3b02d4bb466d53d41d8d58f2d99)
* Copy/pasting using the [keyboard](https://github.com/WordPress/gutenberg/pull/38643/files#diff-d6ad6b9ae7f870e6d8d163e142ababc68c51e42faf92405fba08a3d063bb8eeb)

In a previous iteration I was using `cloneBlock` in that logic to remove unsupported attributes (instead of adding the logic to the serializer), but there was [some concern](https://github.com/WordPress/gutenberg/pull/34750#discussion_r760826441) that we'd have to ensure that logic was utilized any time new functionality was added, particularly by plugins. Given that we're relying on passing a prop to the serializer to exclude those attributes though, I don't see that it's very different. I do think it makes sense to set an expectation that you should explicitly `cloneBlock` a block before duplicating it. 

I'll remove the serializer changes.